### ISSUE TITLE
Some questions about the parameter 'im2col_step' of Deformable ConvNet V2 in the testing process

### ISSUE THREAD

**Issue by crocodilegogogo** [AyushiSinha/medimreg_sc#1]
I have a question about the parameter 'im2col_step' of Deformable ConvNet V2. We used Deformable Conv in classification tasks. We set the training batchsize the same as im2col_step. During the test process, we put different numbers of test samples in test batch (e.g. test the testing dataset by input one sample per time, or test the testing dataset by inputing ten samples per time), and get different classification results. It seems that how many samples we input to the network each time impacts the final classification results. So why is this happening? Will you kindly give me some advice? What's the relationship between testing batchsize and im2col_step? What's the relationship between training batchsize and im2col_step?

**Comment by AyushiSinha** [comment]
That is weird that the final classification results vary. im2col_step should not make a difference - it simply sets the batch size for the im2col operation.

**Comment by crocodilegogogo** [comment]
Thank you for your kind reply. Since the number of test samples may not be integral multiples of the batch size, we cannot simply sets the batch size for the im2col operation. Besides, due to the testing set is huge, we want to accelarate the testing process (need to test per Epoch), thus put more testing samples than the batch size per test. Finally we found this phenomenon.

**Comment by AyushiSinha** [comment]
You may set im2col_step to a size smaller than batch size (maybe just set it to 1). It should be okay to have test size > batch size. Is you bug resolved?

**Comment by AyushiSinha** [comment]
You may set im2col_step to a size smaller than batch size (maybe just set it to 1). It should be okay to have test size > batch size. Is your bug resolved?

**Comment by crocodilegogogo** [comment]
Yep I set im2col_step to 1, and the classification results seem correct. But it's getting obsviously slower...
